controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -1028,7 +1029,7 @@ public Resource createResource(PathAddress relativeAddress) {
     }
 
     private Resource createResourceInternal(PathAddress relativeAddress, int index) throws UnsupportedOperationException {
-        ImmutableManagementResourceRegistration current = getResourceRegistration();
+        ImmutableManagementResourceRegistration current = checkNotNullParamWithNullPointerException("getResourceRegistration", getResourceRegistration());
```

Comment:
Sorry for the back-and-forth @boris-unckel , I missed this one; the purpose of `checkNotNullParamWithNullPointerException` is to check method params, so I would say we should not use it for any kind of situation. If you still want to avoid deferring the NPE here, we could use a specific logger entry for this one.

Reply:
@yersan NP, thanks for the additional feedback. I have changed accordingly.